### ISSUE TITLE
Testing enhanced enums does not effect coverage report

### ISSUE THREAD

**Issue by maxzod** [flutter/flutter#128863]
### Steps to reproduce

# the enum
```dart
enum SignInAuthState {
  canLogin(1),
  mustSignUp(2),
  mustBeInvited(3);

  const SignInAuthState(this.id);

  final int id;
}
```

# command
```bash
flutter test --coverage
```
 

### Expected results

it should cover the enum

### Actual results

the tests does not affect the coverage report

### Code sample

<details><summary>Code sample</summary>
 inside `/test/enums_test.dart `

 

```dart
test(
    'SignInAuthState it return id based on fixed for the server',
    () {
      expect(SignInAuthState.canLogin.id, 1);
      expect(SignInAuthState.mustSignUp.id, 2);
      expect(SignInAuthState.mustBeInvited.id, 3);
    },
  );
 
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console

Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.7.11, on macOS 13.3.1 22E772610a darwin-arm64, locale en-EG)
[!] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.
[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.1)
[✓] VS Code (version 1.79.0)
[✓] Connected device (3 available)
[✓] HTTP Host Availability

! Doctor found issues in 1 category.
 ```

</details>

**Comment by dam-ease** [comment]
Hi @maxzod. By does not effect coverage report, do you mean all tests pass with no code coverage result?

**Comment by maxzod** [comment]
@dam-ease exactly 
the tests pass and if I edit them to fail they fail as intended 
and in both cases, the `coverage/lcov.info` does not indicate that this file was used in the tests 

other code appears to be covered just fine  the problem is to the  enhanced enums tests only

**Comment by dam-ease** [comment]
Hi @maxzod. So, I created a new flutter project using the latest `stable` version. Added the enhanced enums in your code sample to the main.dart file and added your tests as well.
From my log after running `flutter test --coverage`
```
00:01 +1: All tests passed!       
``` 
In my `coverage/lcov.info`, I have:
```
SF:lib/main.dart
DA:6,0
DA:7,0
DA:21,1
DA:23,0
DA:25,0
DA:26,0
DA:27,0
DA:28,0
DA:31,0
DA:33,0
DA:36,0
DA:37,0
DA:38,0
DA:39,0
DA:40,0
LF:15
LH:1
end_of_record

```
Is this your expected behaviour, if yes, you might need to update to the latest `stable` channel to confirm this.
